### PR TITLE
Upgrade to Souffle v2.3 from v2.1.

### PR DESCRIPTION
**Pull request by shaper**
External part of #601

### CODE REVIEW

File: rust/tools/authorization-logic/BUILD [hunk 1]
```diff
@@ -134,11 +134,14 @@ filegroup(
...
 )
 
-# Specifing `@souffle` as part of `data` in the `authorization_logic_test` target
+# Specifying `@souffle` as part of `data` in the `authorization_logic_test` target
 # triggers a build of souffle with `-fno-exceptions`, which causes the build
 # to fail. Instead, we extract the souffle binary using a genrule by making
 # it part of the `tools` (which does not trigger a rebuild) and use it when
 # running the authorization logic tests via `authorization_logic_test`.
+#
+# TODO(wkorman): Investigate warning as below on build:
```
Comment (bgogul) [inline rust/tools/authorization-logic/BUILD:143]: Can we file an issue and use that number in TODO? I think there is very straightforward fix, we should simply change the rule name to something like below:

```
genrule(
  name = "souffle_tool",
  outs = ["souffle"],
   ...
)
```

We will need to change the current uses of souffle though.

**Review by bgogul — APPROVED** [review]
Thanks, Walter!

@markww, I assume we are already out-of-sync with internal souffle that this change is fine?

File: rust/tools/authorization-logic/BUILD (hunk 1 above)
Comment (markww) [reply rust/tools/authorization-logic/BUILD:143]: Yes, that's probably the right solution. And you are correct, we would need to update our usages of `souffle`.

**Comment by markww** [comment]
Yes, no (additional) harm in becoming further out of sync with internal Souffle. But ideally, we should update internal Souffle to also be version 2.3.

File: rust/tools/authorization-logic/BUILD (hunk 1 above)
Comment (shaper) [reply rust/tools/authorization-logic/BUILD]: Done, filed #607

**Review by markww — APPROVED** [review]

**Pull request merged**